firedrake/output.py

```diff
@@ -403,16 +403,16 @@ def __init__(self, filename, project_output=False, comm=None, mode="w",
         self._comm = internal_comm(self.comm)
 
         if self._comm.rank == 0 and mode == "w":
-            outdir = os.path.dirname(os.path.abspath(filename))
-            if not os.path.exists(outdir):
-                os.makedirs(outdir)
+            if not os.path.exists(basename):
+                os.makedirs(basename)
         elif self._comm.rank == 0 and mode == "a":
             if not os.path.exists(os.path.abspath(filename)):
                 raise ValueError("Need a file to restart from.")
         self._comm.barrier()
 
         self.filename = filename
         self.basename = basename
+        self.vtu_basename = os.path.join(basename, os.path.basename(basename))
```

Comment:
I don't understand this line. Why is this the right thing?

Reply:
The `basename` is absolute path to the `.pvd` file without suffix `.pvd`, and it is also the path to `vtu` file. 
Here, we join the `vtu` dir name with the prefix of the `pvd` filename to obtain the `vtu_basename`.

For example, code like this 
```python
output = File('/my/path/to/test.pvd')
output.write(fun)
```
will generate files
```
/my/path/to/test.pvd
/my/path/to/test/test_0.vtu
```
The value of `basename` is  `/my/path/to/test`, and the value of `vtu_basename` is `/my/path/to/test/test`

Reply:
Please provide a comment that explains this. It is not especially obvious.